bson-adt-core/src/main/scala/adt/bson/BsonAdtImplicits.scala

```diff
@@ -0,0 +1,85 @@
+package adt.bson
+
+import adt.bson.mongo.codecs.{BsonAdtCodecRegistry, BsonAdtCodec}
+import org.bson.codecs.configuration.CodecRegistry
+import org.bson.codecs.{DecoderContext, EncoderContext}
+import org.bson.conversions.{Bson => BsonConversion}
+import org.bson.{BsonDocumentReader, BsonDocumentWriter}
+
+import scala.collection.JavaConversions.seqAsJavaList
+import scala.language.{higherKinds, implicitConversions}
+
+/**
+ * Adds implicit conversions between [[JavaBsonValue]] and [[BsonValue]].
+ *
+ * @note these are provided when you `import adt.bson.mongo._`
+ */
+trait BsonAdtImplicits {
+
+  implicit def bsonValueAsJavaOps(adt: BsonValue): BsonAdtValueOps = new BsonAdtValueOps(adt)
```

Comment:
Why not use implicit classes? http://docs.scala-lang.org/overviews/core/implicit-classes.html